libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/NativeLoginManager.kt

```diff
@@ -270,108 +266,194 @@ internal class NativeLoginManager(
         return requestBodyString.toRequestBody(mediaType)
     }
 
-    // region Headless, Password-Less Login Via One-Time-Passcode
+    // region Salesforce Identity API Headless Forgot Password Flow
 
-    override suspend fun submitOtpRequest(
+    override suspend fun startPasswordReset(
         username: String,
         reCaptchaToken: String,
-        otpVerificationMethod: OtpVerificationMethod
-    ): OtpRequestResult {
+    ): NativeLoginResult {
 
         // Validate parameters.
-        if (!isValidUsername(username.trim())) {
-            return OtpRequestResult(InvalidUsername)
-        }
-        val reCaptchaSiteKeyId = when {
-            reCaptchaSiteKeyId == null -> {
-                SalesforceSDKLogger.e(TAG, "A reCAPTCHA site key wasn't and must be provided when using enterprise reCAPATCHA.")
-                return OtpRequestResult(UnknownError)
+        val trimmedUsername = username.trim()
+        trimmedUsername.mapInvalidUsernameToResult()?.let { return it }
+        val reCaptchaParameterGenerationResult = generateReCaptchaParameters(reCaptchaToken)
+
+        // Generate the start password reset request body.
+        val startPasswordResetRequestBodyString = runCatching {
+            StartPasswordResetRequestBody(
+                recaptcha = reCaptchaParameterGenerationResult.nonEnterpriseReCaptchaToken,
+                recaptchaevent = reCaptchaParameterGenerationResult.enterpriseReCaptchaEvent,
+                username = trimmedUsername
+            ).toJson()
+        }.onFailure { e ->
+            SalesforceSDKLogger.e(TAG, "Cannot JSON encode start password reset request body due to an encoding error with message '${e.message}'.", e)
+            return UnknownError
+        }.getOrNull()
+
+        // Create the start password reset request.
+        val startPasswordResetRequest = RestRequest(
+            POST,
+            LOGIN,
+            "$loginUrl$OAUTH_ENDPOINT_HEADLESS_FORGOT_PASSWORD",
```

Comment:
Either way, we ought to be consistent in how strings are assembled.  I lean towards Kotlin's templates since that's what the IDE inspector it guiding the community at large to.  I'll quickly switch if needed here to move this through, though.